pilot/pkg/config/aggregate/config.go

```diff
@@ -107,21 +107,34 @@ func (cr *store) Get(typ config.GroupVersionKind, name, namespace string) *confi
 
 // List all configs in the stores.
 func (cr *store) List(typ config.GroupVersionKind, namespace string) []config.Config {
-	if len(cr.stores[typ]) == 0 {
+	stores := cr.stores[typ]
+	if len(stores) == 0 {
 		return nil
 	}
-	var configs []config.Config
-	// Used to remove duplicated config
-	configMap := sets.New[types.NamespacedName]()
 
-	for _, store := range cr.stores[typ] {
-		storeConfigs := store.List(typ, namespace)
-		for _, cfg := range storeConfigs {
+	var (
+		configs      []config.Config
+		storeConfigs = make([][]config.Config, 0, len(stores))
+		configCnt    int
+	)
+
+	for _, store := range stores {
+		curConfigs := store.List(typ, namespace)
+		storeConfigs = append(storeConfigs, curConfigs)
+		configCnt += len(curConfigs)
+	}
+
+	configs = make([]config.Config, 0, configCnt)
+	// Used to remove duplicated config
+	configMap := sets.NewWithLength[types.NamespacedName](configCnt)
+	for _, curConfigs := range storeConfigs {
+		for _, cfg := range curConfigs {
```

Comment:
we can use slices.FilterInplace here to reduce one loop

Reply:
done